extensions/subclasses/customcontext.py

```diff
@@ -96,37 +100,37 @@ def __init__(self, bot):
 
     async def my_on_message(self, message):
         """This function replaces `bot.on_message`"""
-        # Ignore bots without a heavy check on each command, this kills it right at the message receipt
-        if message.author.bot:
+        if not self._should_reply(message):
             return
-        # A check to see if the message should be ignored
-        for _ in range(1):
-            if str(message.author.id) in self.bot.c.trustedusers:
-                break
-            if message.guild is not None:
-                if str(message.guild.id) in self.bot.c.trustedservers:
-                    break
-                if str(message.guild.id) in self.bot.ignored.guilds:
-                    return
-                if str(message.channel.id) in self.bot.ignored.channels:
-                    return
-            if str(message.author.id) in self.bot.ignored.users:
-                return
-        # now we do the command processing
-        # this little bit allows the bot to process multiple commands in one message, separated by " && "
-        if " && " in message.content:
-            cmds = message.content.split(" && ")
-            for cmd in cmds:
-                message.content = cmd
-                ctx = await self.bot.get_context(message, cls=NoodleContext)
-                if ctx.valid:
-                    await self.bot.invoke(ctx)
-        else:
-            # otherwise, process as normal
-            ctx = await self.bot.get_context(message, cls=NoodleContext)
-            if ctx.valid:
-                await self.bot.invoke(ctx)
 
+        # process multiple commands in one message
+        # NB this prevents " && " from appearing in arguments.
+        # it may also be worth re.splitting on r'\s+&&\s+',
+        # which allows other kinds of whitespace, but reduces performance
+        commands = message.content.split(" && ")
+        for command in commands:
+            sub_message = copy.copy(message)  # prevents some unwanted state mutation
+            sub_message.content = command
+            await self.process_commands(sub_message)
```

Comment:
Foregoing what was had, for `process_commands` appears to break the custom contexter